### PR TITLE
add tests for test package

### PR DESCRIPTION
**Pull request by jgramoll**
also:
• fix ExitStatus for test package
• Allow POST for test package

### CODE REVIEW

File: bridge/bridgetest/bridgetest.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ func MockFunc(e mockEnvironment) net.Conn {
 		for {
 			d, err := readPbFrame(conB)
 			if err != nil {
-				e.Errorf("Can't read method name")
+				// Conn probably closed
```
Comment (jgramoll) [inline bridge/bridgetest/bridgetest.go:122]: I was getting an error with go test -race trying to access e from this goroutine. Open to ideas how to fix this better, but it got my tests working

File: test/test.go [hunk 2]
```diff
@@ -110,13 +110,17 @@ func (req *Request) Validate() error {
...
-	if req.Method == "GET" {
+	switch req.Method {
+	case "GET":
 		if req.Body != "" {
 			return fmt.Errorf("GET requests must not have body, found \"%v\"", req.Body)
 		}
 		return nil
+	case "POST":
+		return nil
+	default:
```
Comment (jgramoll) [inline test/test.go:121]: Is there a reason to only whitelist certain methods?

File: test/test.go [hunk 3]
```diff
@@ -442,9 +446,9 @@ func (e *TestEnv) Handle(method string, args_d []byte) []byte {
 		args := kong_plugin_protocol.ExitArgs{}
 		e.noErr(proto.Unmarshal(args_d, &args))
 		e.ClientRes.Status = int(args.Status)
+		e.state = finished
```
Comment (jgramoll) [inline test/test.go:449]: ExitStatus was not finishing

File: test/test.go (hunk 2 above)
Comment (StarlightIbuki) [reply test/test.go:121]: Looks like unfinished code.

**Comment by kikito** [comment]
@fffonion please review

File: test/test.go (hunk 3 above)
Comment (chobits) [reply test/test.go:449]: ack for this addition

**Comment by StarlightIbuki** [comment]
Hi. @jgramoll Do you still want to push this forward?

**Comment by gszr** [comment]
Opened https://github.com/Kong/go-pdk/pull/196 picking the changes here.

**Pull request closed without merging**